src/ffi/com_model.rs

```diff
@@ -846,204 +252,44 @@ pub struct msurface_s {
-            "::",
-            stringify!(texturemins)
-        )
-    );
-    assert_eq!(
-        unsafe { &(*(null::<msurface_s>())).extents as *const _ as usize },
-        48usize,
-        concat!(
-            "Offset of field: ",
-            stringify!(msurface_s),
-            "::",
-            stringify!(extents)
-        )
-    );
-    assert_eq!(
-        unsafe { &(*(null::<msurface_s>())).texinfo as *const _ as usize },
-        52usize,
-        concat!(
-            "Offset of field: ",
-            stringify!(msurface_s),
-            "::",
-            stringify!(texinfo)
-        )
-    );
-    assert_eq!(
-        unsafe { &(*(null::<msurface_s>())).styles as *const _ as usize },
-        56usize,
-        concat!(
-            "Offset of field: ",
-            stringify!(msurface_s),
-            "::",
-            stringify!(styles)
-        )
-    );
-    assert_eq!(
-        unsafe { &(*(null::<msurface_s>())).samples as *const _ as usize },
-        60usize,
-        concat!(
-            "Offset of field: ",
-            stringify!(msurface_s),
-            "::",
-            stringify!(samples)
-        )
-    );
-    assert_eq!(
-        unsafe { &(*(null::<msurface_s>())).pdecals as *const _ as usize },
-        64usize,
-        concat!(
-            "Offset of field: ",
-            stringify!(msurface_s),
-            "::",
-            stringify!(pdecals)
-        )
-    );
-}
+#[allow(clippy::unnecessary_operation, clippy::identity_op)]
+const _: () = {
+    ["Size of msurface_s"][size_of::<msurface_s>() - 68usize];
+    ["Alignment of msurface_s"][align_of::<msurface_s>() - 4usize];
+    ["Offset of field: msurface_s::visframe"][offset_of!(msurface_s, visframe) - 0usize];
+    ["Offset of field: msurface_s::dlightframe"][offset_of!(msurface_s, dlightframe) - 4usize];
+    ["Offset of field: msurface_s::dlightbits"][offset_of!(msurface_s, dlightbits) - 8usize];
+    ["Offset of field: msurface_s::plane"][offset_of!(msurface_s, plane) - 12usize];
+    ["Offset of field: msurface_s::flags"][offset_of!(msurface_s, flags) - 16usize];
+    ["Offset of field: msurface_s::firstedge"][offset_of!(msurface_s, firstedge) - 20usize];
+    ["Offset of field: msurface_s::numedges"][offset_of!(msurface_s, numedges) - 24usize];
+    ["Offset of field: msurface_s::cachespots"][offset_of!(msurface_s, cachespots) - 28usize];
+    ["Offset of field: msurface_s::texturemins"][offset_of!(msurface_s, texturemins) - 44usize];
+    ["Offset of field: msurface_s::extents"][offset_of!(msurface_s, extents) - 48usize];
+    ["Offset of field: msurface_s::texinfo"][offset_of!(msurface_s, texinfo) - 52usize];
+    ["Offset of field: msurface_s::styles"][offset_of!(msurface_s, styles) - 56usize];
+    ["Offset of field: msurface_s::samples"][offset_of!(msurface_s, samples) - 60usize];
+    ["Offset of field: msurface_s::pdecals"][offset_of!(msurface_s, pdecals) - 64usize];
+};
 #[repr(C)]
-#[derive(Debug, Copy, Clone)]
+#[derive(Debug)]
```

Comment:
```suggestion
#[derive(Debug, Copy, Clone)]
```
Can remain Clone/Copy because no pointers